src/iarc7_motion/iarc_tasks/track_roomba_task.py

```diff
@@ -0,0 +1,168 @@
+            'roombas', OdometryArray, 
+            self._receive_roomba_status)
+
+        try:
+            self._TRANSFORM_TIMEOUT = rospy.get_param('~transform_timeout')
+            self._MIN_MANEUVER_HEIGHT = rospy.get_param('~min_maneuver_height')
+            self._TRACK_HEIGHT = rospy.get_param('~track_roomba_height')
+            self._MAX_TRANSLATION_SPEED = rospy.get_param('~max_translation_speed')
+            self._MAX_ACCEPTABLE_DIST = rospy.get_param('~roomba_max_acceptable_dist')
+        except KeyError as e:
+            rospy.logerr('Could not lookup a parameter for track roomba task')
+            raise
+
+        if self._TRACK_HEIGHT < self._MIN_MANEUVER_HEIGHT:
+            raise ValueError('Track Roomba height was below the minimum maneuver height')
+
+        self._z_holder = HeightHolder()
+
+        self._state = TrackObjectTaskState.init
+
+        while self._roomba_array is None:
+            pass
+
+    def _receive_roomba_status(self, data):
+        self._roomba_array = data
+
+    def get_desired_command(self):
+        if (self._state == TrackObjectTaskState.init or 
+            self._state == TrackObjectTaskState.waiting):
+            self._state = TrackObjectTaskState.track
+
+        if self._canceled:
+            return (TaskCanceled(),)
+
+        if self._roomba_array is None:
+            self._state = TrackObjectTaskState.waiting
+            return (TaskRunning(), NopCommand())
+
+        if self._state == TrackObjectTaskState.track:
+            try:
+                roomba_transform = self._tf_buffer.lookup_transform(
+                                    'level_quad',
+                                    self._roomba_id,
+                                    rospy.Time.now(),
+                                    rospy.Duration(self._TRANSFORM_TIMEOUT))
+            except (tf2_ros.LookupException,
+                    tf2_ros.ConnectivityException,
+                    tf2_ros.ExtrapolationException) as ex:
+                rospy.logerr('ObjectTrackTask: Exception when looking up transform')
+                rospy.logerr(ex.message)
+                return (TaskAborted(msg='Exception when looking up transform during roomba track'),)
+
+            #Creat point centered at drone's center
+            stamped_point = PointStamped()
+            stamped_point.point.x = 0
+            stamped_point.point.y = 0
+            stamped_point.point.z = 0
+
+            #returns point distances of roomba to center point of level quad
+            self._roomba_point =tf2_geometry_msgs.do_transform_point(
+                                                stamped_point, roomba_transform)
+
+            if not self._check_max_roomba_range():
+                return (TaskAborted(msg='The provided roomba is not found or not within a meter of the quad'),)
+
+            if self._check_min_roomba_range():
+                return (TaskDone(),)
+
+            #p-controller
```

Comment:
Sorry, really nitpicky but could you put spaces between the comment sign and the first letter of the comment. You commented consistently in this style everywhere. I think it makes it more readable. (Up for debate)